### ISSUE TITLE
Pump [Bug] 

### ISSUE THREAD

**Issue by andr1976** [DanWBR/dwsim6#234]
**Describe the bug**
When pump gets zero flow in flowsheet the programs fails

**Expected behavior**
Pump duty should be 0 instead of NaN

**Screenshots**
![image](https://user-images.githubusercontent.com/58475535/142937698-73def0e3-bbd3-4236-84ea-9ae9b923dab7.png)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows, 
 - Version [e.g. v6.7.1]

**Comment by andr1976** [comment]
@DanWBR , for the compressor and expander I think part of the fix is wrong, head is independent of flow and should not be zero if there is a pressure difference
```
If CalcMode <> CalculationMode.Head Then
                    AdiabaticHead = 0.0
                    PolytropicHead = 0.0
                End If
```

**Comment by DanWBR** [comment]
@andr1976 so the pressure increases even with zero flow? I'm setting everything to zero so the energy balance doesn't break

**Comment by andr1976** [comment]
@DanWBR say pump is specified with outlet pressure or with a positive dP, even if flow is zero head is positive, but energy is still zero because energy is head times flow

**Comment by DanWBR** [comment]
@andr1976 ok, thanks for the explanation!

**Comment by andr1976** [comment]
@DanWBR , ok I also see your point, that the calculations might break down at some point when power goes to zero, depending on which way you do the math -you also might have to force outlet temperature equal to inlet etc, so it might be just as good just  report an error some how

**Comment by andr1976** [comment]
It works now, but some other issues appear which I am investigating, if this is related to the fix, so keeping this one open for now

**Comment by andr1976** [comment]
Made a fix #241, closing once PR accepted/merged into main branch
